File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Belladonna.cpp
#include <iostream>
#include "Belladonna.h"
#include "World.h"

Belladonna::Belladonna() {
	this->name = 'b';
	this->strength = 99;
}

bool Belladonna::collision(World* world,Organism* organismToFight) {
	//Kills any animal which eats it.
	pos animalPos;
	if (this->isAttacker == true) {
		return true;
	}
	else {
		if (organismToFight->GetName() > LETTER_RANGE_ANIMAL_A && organismToFight->GetName() < LETTER_RANGE_ANIMAL_Z) {
			organismToFight->SetOrganismLifespan(false);
			this->isAlive = false;
			animalPos = organismToFight->GetOrganismPosition();
			world->setOrganismVectorData(nullptr, animalPos.y, animalPos.x);
			world->setOrganismVectorData(nullptr, position.y, position.x);

			world->addCommentatorVector(CONSUMPTION, organismToFight, this, NULL);
			world->addCommentatorVector(KILL, this, organismToFight, NULL);
		}
	}
	return true;
}

char Belladonna::GetName() const {
	return 'b';
}

string Belladonna::GetOrganismFullName() const {
	return "Belladonna";
}

Belladonna::~Belladonna() {

}

File: 2D_sym_python/plants/Belladonna.py
from Plant import*

class Belladonna(Plant):

    def __init__(self):
        super().__init__()
        self.name='b'
        self.strength=99
    
    def collision(self, world, organismToFight):
        if self.isAttacker == True:
            return True
        else:
            if LETTER_RANGE_ANIMAL_A < ord(organismToFight.GetName()) < LETTER_RANGE_ANIMAL_Z:
                organismToFight.SetOrganismLifespan(False)
                self.isAlive = False
                animalPos = organismToFight.GetOrganismPosition()
                world.SetOrganismVectorData(None, animalPos['y'],animalPos['x'])
                world.SetOrganismVectorData(None, self.position['y'],self.position['x'])

                world.AddCommentatorVector(reportType.CONSUMPTION,organismToFight, self,0)
                world.AddCommentatorVector(reportType.KILL, self,organismToFight,0)
        return True
    
    def GetName(self):
        return 'b'
    
    def GetOrganismFullName(self):
        return "Belladonna"
            